### ISSUE TITLE
limit doesn't work with hyper

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy#10802]

**Comment by wrat** [comment]
Ohkkk I looking into this.
@asmeurer I just asking that solving this limit ,As you hint that first we expand it as hyperexpand in limits.py
then solve it by gruntz() funtion ?? or we have implement something ?? as I can fugure out that gruntz is not work properly with funtion whose mrv_leadterm is complex to compute.

**Comment by asmeurer** [comment]
I'm not necessarily suggesting that, I'm just showing what the limit is. I don't know what the best way to fix this issue is, as I'm not very familiar with the limit code.

**Comment by rehassachdeva** [comment]
Hi! Can we do something like this, check if expression is an instance of hyper class, if yes then hyperexpand it before passing as input to gruntz function ?

**Comment by wrat** [comment]
Yeah this is  first  idea , which is comes in my mind but it will  not be a efficient Implementation for gruntz() funtcion . Hope  @asmeurer will suggest something.

**Comment by asmeurer** [comment]
I think the better way to do that is to implement _eval_expand_tractable on hyper. But I don't know if there's a more direct way to make limits work on hyper. If there is, that would be preferable, since not all hyper instances can be converted to other functions with hyperexpand.

**Comment by rehassachdeva** [comment]
Hi. could you please explain.. how  _eval_expand_tractable should work ? I just tried looking into similar `_eval_expand_` methods.

**Comment by rehassachdeva** [comment]
Hi.
As i understood we can implement `_eval_expand_tractable` to return hyperexpand(self). 
But where should we call `e.expand(tractable=true)` ?
Would it be correct to do something like
```
if e.has(hyper):
    e.expand(tractable=True)
```
